narwhals/stable/v1/__init__.py

```diff
@@ -250,6 +253,27 @@ def _l1_norm(self: Self) -> Self:
         """
         return self.select(all()._l1_norm())
 
+    def group_by(  # type: ignore[override]
+        self: Self, *keys: str | Iterable[str], drop_null_keys: bool = False
+    ) -> GroupBy[Self]:
+        r"""Start a group by operation.
+
+        Arguments:
+            *keys: Column(s) to group by. Accepts multiple columns names as a list.
+            drop_null_keys: if True, then groups where any key is null won't be included
+                in the result.
+
+        Returns:
+            GroupBy: Object which can be used to perform aggregations.
+        """
+        if any(isinstance(k, (Expr, Series)) for k in keys):
+            msg = (
+                "Expr and Series are not supported as `narwhals.stable.v1.DataFrame.group_by` keys."
+                "\nUse main namespace class instead: `narwhals.DataFrame.group_by`"
+            )
+            raise ComputeError(msg)
+        return super().group_by(*[col(k) for k in keys], drop_null_keys=drop_null_keys)
```

Comment:
Notice that if we were to do:

```py
return super().group_by(col(*keys), drop_null_keys=drop_null_keys)
```
which in principle is more pythonic/polars-like, then polars `extract_native(arg).meta.output_name()` itself would raise an exception